File: app/src/main/java/com/alexsheiko/invitationmaker/AdProvider.java
package com.alexsheiko.invitationmaker;

import android.content.Context;
import android.widget.Toast;

import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.InterstitialAd;

public class AdProvider {

    private InterstitialAd mInterstitialAd;
    private Context mContext;

    public void prepare(Context context) {
        mContext = context;
        // Initialize the Mobile Ads SDK.
        mInterstitialAd = new InterstitialAd(context);
        mInterstitialAd.setAdUnitId("ca-app-pub-3038649646029056/5392277129");

        requestNewInterstitial();
    }

    private void requestNewInterstitial() {
        AdRequest adRequest = new AdRequest.Builder()
                .addTestDevice(AdRequest.DEVICE_ID_EMULATOR)
                .build();

        mInterstitialAd.loadAd(adRequest);
    }

    public void showImage() {
        if (mInterstitialAd.isLoaded()) {
            mInterstitialAd.show();
        } else {
            Toast.makeText(mContext, "Retry in 2 seconds", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/alexsheiko/invitationmaker/util/AdProvider.kt
package com.alexsheiko.invitationmaker.util

import android.content.Context
import com.google.android.gms.ads.AdListener
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.InterstitialAd

class AdProvider(mContext: Context) {

    private var mInterstitialAd: InterstitialAd = InterstitialAd(mContext)

    init {
        mInterstitialAd.adUnitId = "ca-app-pub-3038649646029056/5392277129"
    }

    fun loadInBackground() {
            requestNewInterstitial()
    }

    private fun requestNewInterstitial() {
        val adRequest = AdRequest.Builder()
                .addTestDevice(AdRequest.DEVICE_ID_EMULATOR)
                .addTestDevice("1A6B43A15E989B8B4F9121A9D649E323")
                .build()
        mInterstitialAd.loadAd(adRequest)
    }

    fun show() {
        if (mInterstitialAd.isLoaded) {
            mInterstitialAd.show()
        } else {
            mInterstitialAd.adListener = object : AdListener() {
                override fun onAdLoaded() {
                    mInterstitialAd.show()
                    mInterstitialAd.adListener = null
                }
            }
        }
    }
}
